### DISCUSSION TITLE
Table of content also show tabs

### DISCUSSION THREAD

**Discussion by SPyofgame** [facelessuser/pymdown-extensions#1992] [Q&A]
There are some specific cases where I am working with large content and I want to enhance their organization by creating sub-content for specific sections by using tabs. However, these sections upon division did share certain properties I want to mention in the table of content. Can I add some specifications to those tabs in order to make them appear on [TOC] ?

For example:

!!! example

    === "Unordered List" { tab-name = "Unordered List" }

        ``` markdown
        * Sed sagittis eleifend rutrum
        * Donec vitae suscipit est
        * Nulla tempor lobortis orci
        ```

    === "Ordered List" { tab-name = "Ordered List" }

        ``` markdown
        1. Sed sagittis eleifend rutrum
        2. Donec vitae suscipit est
        3. Nulla tempor lobortis orci
        ```
        
        
    === "Something that is not important that is by default won't be shown in the TOC"

        ``` c++
        #include <iostream>
    
        int main(void) {
          std::cout << "Hello world!" << std::endl;
          return 0;
        }
        ```

**Comment by SPyofgame** [comment]
Oh, I found a workaround using this

```
#### <p hidden href="..."> Unordered List </p>
```

**Reply by SPyofgame** [reply]
Is there the other way without manually adding the specific href for each content? :<

**Reply by facelessuser** [reply]
I'm not really understanding the question fully. Is this a MkDocs question? MkDocs only adds headers to the TOC. There may be a way to inject things into the TOC maybe with a MkDocs plugin, but this is mainly a repo of Python Markdown extensions, which directly affect the Markdown parser, not the higher-level MkDocs work flow.

**Reply by SPyofgame** [reply]
Sorry for my confusing request. 

So here is an example: I want to write a wiki for algorithms

For my specific article, I want to describe the basic properties of my article in 4 separate panels as follows:

![image](https://user-images.githubusercontent.com/62207095/226251673-a260dc75-f4a5-43cf-a799-e8479d19a417.png)

But I also want the table of content will also address the link to the important panels, fox examples for these three only:

![image](https://user-images.githubusercontent.com/62207095/226251880-dd8ba04f-62d3-40f5-a013-60e22f717f03.png)

Those are useful for beginners, however, the audience might be more advanced and should be able to skip them easily and conveniently. Hence I put them in tabs instead of writing using a header, but they can only anchor one tab at a time (even when I use `#### <p hidden> fake toc tab </p>`)

![image](https://user-images.githubusercontent.com/62207095/226252448-21ae4829-da33-4585-a753-1e57751da483.png)

Here is the code

```md

[TOC]

----
## Thông tin cơ bản (Basic Properties)



=== "Giới thiệu (Introduction)" 

    #### <p hidden> Giới thiệu </p>
    Quy hoạch động tổng tập con, là một chủ đề khá mới nhưng đang dần phổ biến ở Việt Nam trong kỳ thi học sinh giỏi quốc gia môn tin học (VOI) khi lần đầu được xuất hiện ở [bài 3][VOI2020_3] VOI 2020. Tất nhiên nó cũng đã xuất hiện trước đó nhiều năm trước như kì thi ICPC Amritapuri Regionals 2014.

    [VOI2020_3]: https://oj.vnoi.info/problem/voi20stars "CÁC CHÒM SAO (6 điểm)"
     

=== "Định nghĩa (Definitions)"

    #### <p hidden> Định nghĩa </p>

    Cho một mảng $a[n]$ với độ dài cố định [$n = 2^k$][ref], mỗi phần tử là một số nguyên.
    
    [ref]: https://es.ac.vn "Có thể không cần là chính xác luỹ thừa của $2$, nhưng đặt thế để tiện tính toán và tránh những lỗi sai không đáng có."

    Ta cần tính được mảng $F[mask] = \underset{sub \subseteq mask}{\LARGE \Sigma} a[sub]$.
    
=== "Đối tượng độc giả (Target of Audience)"
    
    #### <p hidden> Đối tượng độc giả </p>

    Những kiến thức cơ bản bạn nên biết qua để có thể hiểu được bài học này:   
    - [DP][DP]: Quy hoạch động. (1)   
    - [Bitmask][Bitmask]: Biểu diễn trạng thái tập hợp bằng bit. (2)   
    { .annotate }

    1. ==Quy hoạch động==: Là một kỹ thuật giúp đơn giản hóa hiệu quả việc chia bài toán lớn thành các bài toán con chồng chéo nhau hay có công thức truy hồi, mà ở đó không cần tính lại nhiều lần cho cùng một bài toán.
    2. ==Biểu diễn trạng thái tập hợp bằng bit==: Là số nguyên biểu diễn một tập số bằng các bit {0,1} của nó, mà ở đó bit thứ $p$ là $1$ khi và chỉ khi tập hợp chứa phần tử $p$.


    Khuyến khích với các bạn đủ chuyên sâu để làm được các bài tập vận dụng nâng cao quy hoạch động trạng thái ([dp bitmask](https://codeforces.com/blog/entry/81516)).
    
    [DP]: https://en.wikipedia.org/wiki/Dynamic_programming "Là viết tắt của Dynamic Programming, hay còn gọi là quy hoạch động."
    
    [Bitmask]: https://es.ac.vn/bitmask "Là kĩ thuật biểu diễn tập hợp bằng các bit của số nguyên"

    

=== "Tác giả bài viết (Article Author)"

    #### <p hidden> Tác giả bài viết </p>

    ✍🏻 Writer: [@SPyofgame](https://hackmd.io/@SPyofgame)   
    🔎 Editor:  
    👀 Contributor: [@ngpin04](https://hackmd.io/@ngpin04)   
    📋 Cost: 10 hours.
```

**Reply by facelessuser** [reply]
You can actually make tabs generate an ID with the slugify option, but the TOC extension only searches headers. The TOC extension would be responsible for that. I don't know how easy it would be to inject such IDs into the TOC, but I imagine someone could craft and extension to attempt injecting other existing IDs into the TOC.

**Reply by SPyofgame** [reply]
I did try something with <href> with <#ID> and similar but the result isn't good enough 😢 

Either this side-navigation table of content
![image](https://user-images.githubusercontent.com/62207095/226275390-7b3bedad-5940-4f54-bcff-8066a85715db.png)

or this [TOC] inside the content
![image](https://user-images.githubusercontent.com/62207095/226275444-0f692be7-addd-4ce6-a6de-d9bf03aca3f6.png)

still does not work well nor being convenient. 

Do you have another recommendation?

**Reply by facelessuser** [reply]
I don't control TOC generation you can generate IDs for tabs and manually link them, but TOC generation is controlled by a whole different process. TOC only looks for headers. Someone would have to create a whole different TOC implementation that creates the TOC from IDs in headers _and_ other elements. Or, they would somehow have to inject into the TOC new entries after it was built. So currently, there is no solution that I know of at this time.

**Reply by SPyofgame** [reply]
😿 oh, so unfortunate hearing that.

May I ask that will there be certain updates or features that allow me to do this easier @squidfunk ?

**Reply by squidfunk** [reply]
MkDocs generates the table of contents from headers, not from content tabs. Any extension of Markdown processing would need to be added either to Python Markdown, Python Markdown Extensions or MkDocs. I'm solely maintaining Material for MkDocs, which only consumes what it gets from those three libraries. Yes, we also ship some plugins, but we don't do Markdown processing.

**Reply by SPyofgame** [reply]
Oh, ok then 😞

**Reply by SPyofgame** [reply]
Wait, so it is possible with [this](https://facelessuser.github.io/pymdown-extensions/extensions/tabbed/) ?

![image](https://user-images.githubusercontent.com/62207095/226993575-e877ac8d-a081-4bb4-8e41-9a5d73fa2fdc.png)

**Reply by facelessuser** [reply]
This creates IDs, it doesn't inject them into the TOC

**Reply by SPyofgame** [reply]
oh 😭

**Reply by SPyofgame** [reply]
Wait, it means I can not jump into any link whose tabs aren't ready? I just tested 😢 

This will be a huge disadvantage for me when I want to represent different algorithms in separate tabs 😭

**Reply by facelessuser** [reply]
IIRC, Material allows you to jump to a tab with an ID, there is just no TOC.
